packages/cli/src/manual-execution.service.ts

```diff
@@ -0,0 +1,124 @@
+import * as a from 'assert/strict';
+import {
+	DirectedGraph,
+	filterDisabledNodes,
+	recreateNodeExecutionStack,
+	WorkflowExecute,
+} from 'n8n-core';
+import type {
+	IPinData,
+	IRun,
+	IRunExecutionData,
+	IWorkflowExecuteAdditionalData,
+	IWorkflowExecutionDataProcess,
+	Workflow,
+} from 'n8n-workflow';
+import type PCancelable from 'p-cancelable';
+import { Service } from 'typedi';
+
+import { Logger } from '@/logging/logger.service';
+
+@Service()
+export class ManualExecutionService {
+	constructor(private readonly logger: Logger) {}
+
+	getExecutionStartNode(data: IWorkflowExecutionDataProcess, workflow: Workflow) {
+		let startNode;
+		if (
+			data.startNodes?.length === 1 &&
+			Object.keys(data.pinData ?? {}).includes(data.startNodes[0].name)
+		) {
+			startNode = workflow.getNode(data.startNodes[0].name) ?? undefined;
+		}
+
+		return startNode;
+	}
+
+	// eslint-disable-next-line @typescript-eslint/promise-function-async
```

Comment:
I'd really like to get rid of `PCancelable` so that we can stop making exceptions like these.